lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -79,7 +80,14 @@ export async function generateLockFile(
       constraint:
         config.constraints?.npm ??
         getPackageManagerVersion('npm', await lazyPkgJson.getValue()) ??
-        (await getNpmConstraintFromPackageLock(lockFileDir)) ??
+        (await getNpmConstraintFromPackageLock(
+          lockFileDir,
+          'package-lock.json',
+        )) ??
+        (await getNpmConstraintFromPackageLock(
+          lockFileDir,
+          'npm-shrinkwrap.json',
+        )) ??
```

Comment:
```suggestion
        (await getNpmConstraintFromPackageLock(lockFileDir, filename)) ??
```

Would this work instead?

Reply:
Yes, that works!